engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -62,17 +62,28 @@ func CreateEnclavePool(
 		)
 	}
 
-	// The amount of idle enclaves is equal to the chan capacity + one enclave (this one will be waiting in the queue until the channel is unblocked)
-	// chanCapacity = 0  means that it will be an unbuffered, so the communication between the channels will succeed when both (sender and receiver) are ready
-	chanCapacity := poolSize - oneEnclave
+	// this channel is the repository of idle enclave UUIDs
```

Comment:
Do we need a channel for a simple "repository", as you say? Especially a buffered array, those are complex to reason about (need to think of when a call will be blocking, when it's not, etc), it adds complexity to the code and if we don't need such a structure, let's use something simpler. 
What about a simple list? You can add an element with a simple `append`, and to remove one, just do something like:
```
if len(idleEnclavesUuids) > 0 {
    enclaveUuid = idleEnclavesReservoir[0]
    idleEnclavesUuids = idleEnclavesUuids[1:]
    return enclaveUuid
}
```

Reply:
I think that we need a channel because the `get` and `fill` methods are executed in different routines, if we use a list we could have race conditions when we append a new element from both sides. I think that a channel is better in this case, but we can discuss it, probably I could missing something. I even tried to add a FIFO queue with a list but then I realized that I should add a mutex or something to avoid race conditions which will block the client from getting an enclave from the pool.

Reply:
Got it got it, yeah using a channel might be easier than a list and mutex. I'm in sync with you here